### PR TITLE
fix(search): remove useless observed at from temporal entity evolution response #236

### PR DESCRIPTION
**Pull request by philippestoltz**

### CODE REVIEW

File: search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt [hunk 1]
```diff
@@ -252,7 +252,8 @@ class TemporalEntityAttributeService(
                     instanceToEnrich.putIfAbsent(JSONLD_TYPE, NGSILD_PROPERTY_TYPE.uri)
                     // remove the existing value as we will inject our list of results in the property
                     instanceToEnrich.remove(NGSILD_PROPERTY_VALUE)
-
+                    // remove the observedAt field as its values will be contained in the values array
+                    instanceToEnrich.remove(NGSILD_OBSERVED_AT_PROPERTY)
```
Comment (HoucemKacem) [inline search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt:256]: Can you try with this syntax ?
`instanceToEnrich.keys.removeAll(setOf(NGSILD_PROPERTY_VALUE, NGSILD_OBSERVED_AT_PROPERTY))`

Comment (philippestoltz) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt]: OK, done

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt [hunk 2]
```diff
@@ -263,27 +264,33 @@ object JsonLdUtils {
...
             inputToMap(input)
         } else {
-            val includedKeys = JSONLD_ENTITY_MANDATORY_FIELDS.plus(includedAttributes)
-            inputToMap(input).filterKeys { includedKeys.contains(it) }
+            val mandatoryFields = if (isExpandedForm)
+                JSONLD_EXPANDED_MANDATORY_FIELDS
+            else
+                JSONLD_COMPACTED_MANDATORY_FIELDS
+            val includedKeys = mandatoryFields.plus(includedAttributes)
+            val tmp = inputToMap(input)
```
Comment (bobeal) [inline shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:292]: this `tmp` variable could be avoided

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt [hunk 3]
```diff
@@ -48,8 +48,9 @@ object JsonLdUtils {
     const val JSONLD_ID = "@id"
     const val JSONLD_TYPE = "@type"
     const val JSONLD_VALUE_KW = "@value"
-    val JSONLD_ENTITY_MANDATORY_FIELDS = setOf("id", "type", "@context")
-    val JSONLD_ENTITY_CORE_PROPERTIES = setOf("id", "type", "@context")
+    const val JSONLD_CONTEXT = "@context"
+    val JSONLD_EXPANDED_MANDATORY_FIELDS = setOf(JSONLD_ID, JSONLD_TYPE, JSONLD_CONTEXT)
+    val JSONLD_COMPACTED_MANDATORY_FIELDS = setOf("id", "type", JSONLD_CONTEXT)
```
Comment (bobeal) [inline shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:53]: I would leave the `Entity` word in the variable name as there are mandatory fields for other NGSI-LD data types

**Comment by bobeal** [comment]
In the PR description, reference the issue with a #<issue number> instead of closes <issue number> (and you can also link the issue to the PR if you want automatic closing when the PR is merged)

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt (hunk 2 above)
Comment (philippestoltz) [reply shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:292]: yes

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt (hunk 3 above)
Comment (philippestoltz) [reply shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:53]: ok

**Review by bobeal — APPROVED** [review]

**Pull request merged**